src/transformers/pipelines/text_generation.py

```diff
@@ -242,7 +242,12 @@ def _forward(self, model_inputs, **generate_kwargs):
 
         # If there is a prefix, we may need to adjust the generation length. Do so without permanently modifying
         # generate_kwargs, as some of the parameterization may come from the initialization of the pipeline.
+        streamer = None
+        if "streamer" in generate_kwargs.keys():
+            streamer = generate_kwargs.pop("streamer")
         generate_kwargs = copy.deepcopy(generate_kwargs)
+        if streamer:
+            generate_kwargs["streamer"] = streamer
```

Comment:
```suggestion
        generate_kwargs = copy.copy(generate_kwargs)
```

I think we can make it a shallow copy (see [this comment](https://github.com/huggingface/transformers/issues/23552#issuecomment-1584205325))